content/blog/2023/07/camel-k-2-0/index.md

```diff
@@ -0,0 +1,193 @@
+
+# Remove deprecated CLI commands
+
+As we embarked in a major upgrade, we took the opportunity to clean all the **deprecated CLI commands** and configuration we had marked in Camel K version 1.
+
+* `kamel install --kaniko-build-cache`
+* `kamel local`
+* `kamel init`
+* `kamel run --config|--resource file:`
+
+In version 2 they won't be any longer available and the operator may fail if those configurations are provided. Above all, the last one will require you to provide a resource file as either a Configmap or Secret if you want to use the related configuration (we don't support any longer the auto-generated Confimaps from file).
+
+# Multiple data types and schemas in Kamelets
+
+Kamelets are now able to expose supported input/output data types as part of the [Kamelet data type specification](/camel-k/next/kamelets/kamelets-dev.html#_kamelet_data_types) in order to increase the interoperability with other components. When referencing a Kamelet as a source or sink in a Pipe users may now choose an appropriate input/output data type that fits best for their individual use case.
+
+This feature leverages the [data types transformer concept](/manual/transformer.html) in Apache Camel 4 and reveals its full power once the new major version of Camel is available (we are all eagerly awaiting it).
+
+# Default Micrometer for integration monitoring
+
+About observability we made some work to use **Micrometer as a default**. Some changes are expected in the default metrics as it replaced the dependencies used from one technology (Microprofile) to another (Micrometer).
+
+You can have a look at the [Prometheus trait configuration](/camel-k/next/traits/prometheus.html) to learn more.
+
+# Multi architecture support
+
+This feature should not be considered as generally available but more a kind of "tech preview". In Camel K 2.0 we provide an **ARM64 architecture based container image** for the operator but it is not tested due to the difficulty to find the required infrastructure and have the suite of test running. It's a best effort feature that we hope we can make generally available in the future versions.
+
+For more information you can have a look at the [Multi architecture official documentation](/camel-k/next/installation/advanced/multi-architecture.html) page.
+
+# Code coverage report
+
+If you're a Camel K contributor you have noticed that we introduced a nice action which is taking care to calculate the code coverage of the new code developed. The actions reports the coverage difference, so we know how we increase (or decrease) indirectly the quality of the code. This is not really (yet) a gateway, we want to create awareness among Camel K developers about the level of quality we introduce with each change. Little by little we expect to **increase the quality of the code** we produce and have less bugs affecting our code.
+
```

Comment:
```suggestion
# Optimize operator performance and resource allocation

The team has made some effort to reduce the amount of resources (memory and CPU) consumed by the Camel K operator. The results of this can be seen when creating many Integrations on a cluster and when the operator runs multiple integration builds.

We have run some test scenarios and compared the results of different Camel K versions.

_Test scenario: Running 600 Camel K integrations_

| Camel K | avg. Memory | avg. CPU | Time |
| :------- | :------------: | :--------: | :----: |
| 1.9.0 | 529 MiB  | 262 millicores | 29 mins |
| 1.12.1 | 344 MiB | 54 millicores | 21 mins |
| 2.0.0 | 306 MiB | 60 millicores | 20 mins |

_Test scenario: Running 50 integrations builds in parallel_

| Camel K | avg. Memory | avg. CPU | Time |
| :------- | :------------: | :--------: | :----: |
| 1.9.0 | 510 MiB | 352 millicores | 59 mins |
| 1.12.1 | 440 MiB | 185 millicores | 57 mins |
| 2.0.0 | 326 MiB | 45 millicores | 38 mins |

The Camel K 2.0 results are pleasant especially when running multiple builds at the same time. The team is keen to keep improving the operator performance in future releases.
```